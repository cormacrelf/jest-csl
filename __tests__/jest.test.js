const { jestCSL, TestEngine, readTestUnits } = require('../index');
const config = require('./integration.test');

jestCSL(config);

const { jestTestCase } = require('../src/jest');
const engine = new TestEngine(config);
const failing = readTestUnits(['./__tests__/failing.yaml']);

describe("programmatic test unit", () => {
  it("should fail a test", () => {
    const failingTest = failing[0].tests.find(t => t.it === 'should fail a single test');
    expect(() => jestTestCase(engine, failingTest)).toThrowErrorMatchingSnapshot();
  });
  it("should fail a sequence", () => {
    const failingTest = failing[0].tests.find(t => t.it === 'should fail a sequence');
    expect(() => jestTestCase(engine, failingTest)).toThrowErrorMatchingSnapshot();
  });
})
