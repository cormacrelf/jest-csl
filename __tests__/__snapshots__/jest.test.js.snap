// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`programmatic test unit should fail a sequence 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).toMatchObject([22m[32mexpected[39m[2m)[22m

Expected value to match object:
  [32m[\\"(DoeNOT)\\"][39m
Received:
  [31m[\\"(Doe)\\"][39m
Difference:
[32m- Expected[39m
[31m+ Received[39m

[2m  Array [[22m
[32m-   \\"(DoeNOT)\\",[39m
[31m+   \\"(Doe)\\",[39m
[2m  ][22m"
`;

exports[`programmatic test unit should fail a test 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).toBe([22m[32mexpected[39m[2m) // Object.is equality[22m

Expected: [32m\\"(DoeNOT)\\"[39m
Received: [31m\\"(Doe)\\"[39m"
`;
